### PR TITLE
Use sample attributes

### PR DESCRIPTION
**Pull request by damien-lloyd**
# Motivation and Context
Sample attributes for social surveys contain information regarding a respondent such as address, contact details and so forth. We are exposing these sample attributes in the sample service so that they can be exposed to other web services. In order to expose sample attributes, we need to expose the sample unit ID (UUID) which also needs to be propagated to other web services. This is because a sample unit ID is unique, whereas sample unit refs are not unique for social surveys.

# What has changed
Tests have been written to ensure sample attributes are exposed as required. Sample attributes now have an endpoint to be retrieved by sample unit id.

# How to test?

1. You need to build rm-samplesvc-api before building rm-sample-service
2. Run all the tests in SampleEndPointIT and SampleEndpointUnitTest in rm-sample-service

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java [hunk 1]
```diff
@@ -125,4 +128,19 @@ public void verifyBadBindingResultThrowsException() throws Exception {
...
+//      SampleAttributes sampleAttribs = new SampleAttributes();
+//      Map<String, String> attribs = new HashMap<>();
+//
+//      attribs.put("ref", id.toString());
+//      sampleAttribs.setAttributes(attribs);
+//
+//      when(sampleService.findSampleAttributes(any())).thenReturn(sampleAttribs);
+//
+//      sampleEndpoint.requestSampleAttributes(id);
+    }
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java]: Entire test is commented out, should be removed or implemented.

File: src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java [hunk 2]
```diff
@@ -131,10 +135,13 @@ public SampleSummary ingest(final MultipartFile file, final String type) throws
   }
 
   @RequestMapping(value = "/{type}/fileupload", method = RequestMethod.POST, consumes = "multipart/form-data")
-  public final @ResponseBody ResponseEntity<SampleSummary> uploadSampleFile(@PathVariable("type") final String type, @RequestParam("file") MultipartFile file) throws CTPException {
+  public final @ResponseBody
+  ResponseEntity<SampleSummary> uploadSampleFile(@PathVariable("type") final String type, @RequestParam("file") MultipartFile file) throws CTPException {
     log.info("Entering Sample file upload for Type {}", type);
-    if (!Arrays.asList("B", "CENSUS", "SOCIAL").contains(type.toUpperCase())) {
-      return ResponseEntity.badRequest().build();
+    if (!Arrays.asList("B", "CENSUS", "SOCIAL")
```
Comment (emilioward) [inline src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: Is there any reason why its called `B` rather than `Business`?

Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: Not that I can see - it was originally built like that.   If we change it now we'll need to change it in response-operations-ui etc

Comment (emilioward) [reply src/main/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpoint.java]: No thats fine I was just curious more than anything else 👍

File: src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java (hunk 1 above)
Comment (damien-lloyd) [reply src/test/java/uk/gov/ons/ctp/response/sample/endpoint/SampleEndpointUnitTest.java]: I've added a unit test which will cover getting sample attributes from a sample unit

**Comment by madeye-matt** [comment]
@AdamHawtin PR comments have been addressed - can you take another look? Ta

File: src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterSocial.java [hunk 3]
```diff
@@ -69,6 +69,8 @@ public SampleSummary ingest(final SampleSummary sampleSummary, final MultipartFi
     private SocialSampleUnit parseLine(CSVRecord line) throws CTPException{
         SocialSampleUnit sampleUnit = new SocialSampleUnit();
         sampleUnit.setAttributes(line.toMap());
+        String ref = (String) line.toMap().get("Reference");
+        sampleUnit.setSampleUnitRef(ref);
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterSocial.java]: Is casting to String necessary here? Also I don't think any of the tests check the sampleUnitRef is set correctly here.

**Pull request merged**